x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportMigrateToDataTiersAction.java

```diff
@@ -137,7 +139,16 @@ public void onFailure(Exception e) {
 
             @Override
             public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
-                super.clusterStateProcessed(oldState, newState);
+                clusterService.getRerouteService()
+                    .reroute("cluster migrated to data tiers routing", Priority.NORMAL, new ActionListener<ClusterState>() {
+                        @Override
+                        public void onResponse(ClusterState clusterState) {}
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            logger.warn("unsuccessful reroute after migration to data tiers routing", e);
```

Comment:
Nit: if `MasterService#isPublishFailureException(e)` this message should just be at `DEBUG`, no need to make more noise on a master failure than necessary.

Reply:
TIL